signer/src/emily_client.rs

```diff
@@ -159,6 +171,41 @@ impl EmilyInteract for EmilyClient {
             .map_err(EmilyClientError::UpdateDeposits)
             .map_err(Error::EmilyApi)
     }
+
+    async fn confirm_deposit(
+        &self,
+        deposit: &SweptDepositRequest,
+        sbtc_txid: &StacksTxId,
+        stacks_chain_tip: &StacksBlock,
+        btc_fee: bitcoin::Amount,
+    ) -> Result<UpdateDepositsResponse, Error> {
+        let update_request = DepositUpdate {
+            bitcoin_tx_output_index: deposit.output_index,
+            bitcoin_txid: deposit.txid.to_string(),
+            status: Status::Confirmed,
+            last_update_block_hash: stacks_chain_tip.block_hash.to_string(),
+            last_update_height: stacks_chain_tip.block_height,
+            fulfillment: Some(Some(Box::new(Fulfillment {
+                bitcoin_txid: deposit.sweep_txid.to_string(),
+                // TODO: we don't keep a mapping between sweep tx inputs and
+                // deposits requests fulfilled, should we?
+                bitcoin_tx_index: 0,
```

Comment:
> // TODO: we don't keep a mapping between sweep tx inputs and
> // deposits requests fulfilled, should we?

Well, we implicitly do since we store the raw transaction. We can compute the index here using the `sweep_tx` field of the `SweptDepositRequest` object, which holds the bitcoin transaction. Does this make sense?